pal/input/web/hmd-input.ts

```diff
@@ -56,20 +116,31 @@ export class HMDInputDevice {
         this._eventTarget.on(eventType, callback, target);
     }
 
+    private _updateNativePoseState (info: PoseInfo) {
+        if (info.code !== Pose.VIEW_LEFT && info.code !== Pose.VIEW_RIGHT && info.code !== Pose.HEAD_MIDDLE) {
+            return;
+        }
+
+        this._nativePoseState[info.code] = {
+            position: new Vec3(info.position.x, info.position.y, info.position.z),
+            orientation: new Quat(info.orientation.x, info.orientation.y, info.orientation.z, info.orientation.w),
+        };
+    }
+
     private _initInputSource () {
         this._viewLeftPosition = new InputSourcePosition();
-        this._viewLeftPosition.getValue = () => Vec3.ZERO;
+        this._viewLeftPosition.getValue = () => this._nativePoseState[Pose.VIEW_LEFT].position;
```

Comment:
What's native pose sate? What happen if not run on native platforms?

Reply:
The variable name is not named correctly, it is a web input pose